utils/_context/virtual_machines.py

```diff
@@ -331,3 +330,18 @@ def __init__(self, **kwargs) -> None:
             os_cpu="amd64",
             **kwargs,
         )
+
+
+class Fedora39amd64(_VirtualMachine):
+    def __init__(sef, **kwargs) -> None:
+        super().__init__(
+            "Fedora_39_amd64",
+            aws_config=_AWSConfig(ami_id="ami-0ba8f2988373d0735", ami_instance_type="t2.medium", user="fedora"),
```

Comment:
I think you are using the worng AMI id. See this message running the tests:
`06:21:01.786 INFO     Diagnostics:
06:21:01.786 INFO       aws:ec2:Instance (Fedora_39_amd64):
06:21:01.786 INFO         error: 1 error occurred:
06:21:01.786 INFO         	* creating EC2 Instance: OptInRequired: In order to use this AWS Marketplace product you need to accept terms and subscribe. To do so please visit https://aws.amazon.com/marketplace/pp?sku=ec8t8q98ms0b0idcq74tlco2a
06:21:01.786 INFO         	status code: 401, request id: 23227fff-a900-4042-8dfa-426554cecfce
06:21:01.786 INFO     
06:21:01.786 INFO       pulumi:pulumi:Stack (system-tests-vms-system-tests_onboarding):
06:21:01.786 INFO         error: update failed`

Pipeline: https://gitlab.ddbuild.io/DataDog/system-tests/-/pipelines/44748391

Try to find the correct ami from: aws console --> ec2 --> images --> amis --> search for public images

Reply:
Every other AMI I use here gives me this message in the logs
```
command:remote:Command -Fedora_39_amd64-init-config creating (15s) Dial 1/inf failed: retrying
```